src/sympc/session/session.py

```diff
@@ -185,41 +185,98 @@ def get_protocol(self) -> Protocol:
         """
         return self.protocol
 
-    def przs_generate_random_share(
+    def _generate_random_share(
         self,
         shape: Union[tuple, torch.Size],
-    ) -> Any:
-        """Generate a random share using the two generators hold by a party.
+    ) -> List[torch.Tensor]:
+        """Generate random tensor share for the given shape.
 
         Args:
             shape (Union[tuple, torch.Size]): Shape for the share.
 
         Returns:
-            Any: ShareTensor
-
+            List[torch.Tensor] : shares generated by the generators.
         """
-        from sympc.tensor import ShareTensor
-
         gen0, gen1 = self.przs_generators
 
-        current_share = generate_random_element(
+        tensor_share1 = generate_random_element(
             tensor_type=self.tensor_type,
             generator=gen0,
             shape=shape,
         )
 
-        next_share = generate_random_element(
+        tensor_share2 = generate_random_element(
             tensor_type=self.tensor_type,
             generator=gen1,
             shape=shape,
         )
+        # We also invoke second generator for prrs as it operates in CTR mode.
+        return [tensor_share1, tensor_share2]
 
-        # It has encoder_precision = 0 such that the value would not be encoded
-        share = ShareTensor(
-            data=current_share - next_share,
-            session_uuid=self.uuid,
-            config=Config(encoder_precision=0),
-        )
+    def przs_generate_random_share(
+        self,
+        shape: Union[tuple, torch.Size],
+    ) -> Any:
+        """Generate a random zero share using the two generators hold by a party.
+
+        Args:
+            shape (Union[tuple, torch.Size]): Shape for the share.
+
+        Returns:
+            Any: ShareTensor or ReplicatedSharedTensor
+
+        """
+        from sympc.tensor import ReplicatedSharedTensor
+        from sympc.tensor import ShareTensor
+
+        current_share, next_share = self._generate_random_share(shape)
+
+        if self.protocol.share_class == ShareTensor:
+            # It has encoder_precision = 0 such that the value would not be encoded
+            share = ShareTensor(
+                data=current_share - next_share,
+                session_uuid=self.uuid,
+                config=Config(encoder_precision=0),
+            )
+        else:
+            share = ReplicatedSharedTensor(
+                shares=[current_share - next_share],
+                session_uuid=self.uuid,
+                config=Config(encoder_precision=0),
+            )
+        return share
+
+    def prrs_generate_random_share(
+        self,
+        shape: Union[tuple, torch.Size],
+    ) -> Any:
```

Comment:
Why do we not have return type as `Union[ReplicatedSharedTensor, ShareTensor]`?

Reply:
I missed it,Sure ,I would modify it :+1:

Reply:
It was in the previous PRZS code, as we import module inside the function so we  have used `Any` .It was placed inside functions to avoid circular imports.I tried solving circular imports ,but had to make a lot of changes ,thought to defer it for a later PR.